hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/AdaptedMonoProcessLogic.java

```
@@ -0,0 +1,87 @@
+        this.preHandleWorkflow = preHandleWorkflow;
+        this.monoProcessLogic = monoProcessLogic;
+    }
+
+    @Override
+    public void incorporateConsensusTxn(final ConsensusTransaction platformTxn, final long submittingMember) {
+        if (platformTxn.getMetadata() instanceof TransactionMetadata metadata) {
+            final var accessor = adaptForMono(platformTxn, metadata);
+            platformTxn.setMetadata(accessor);
+        }
+        monoProcessLogic.incorporateConsensusTxn(platformTxn, submittingMember);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private SwirldsTxnAccessor adaptForMono(
+            final ConsensusTransaction platformTxn, final TransactionMetadata metadata) {
+        try {
+            final var accessor = PlatformTxnAccessor.from(platformTxn.getContents());
+            // TODO - recompute required keys and compare with metadata
+            accessor.addAllCryptoSigs(metadata.cryptoSignatures());
+            final var preHandleStatus = metadata.status();
+            final var payerKey = metadata.payerKey();
+            if (payerKey != null) {
+                if (preHandleStatus != OK) {
+                    accessor.setSigMeta(forPayerOnly((JKey) payerKey, metadata.cryptoSignatures(), accessor));
+                } else {
+                    accessor.setSigMeta(forPayerAndOthers(
+                            (JKey) payerKey, (List) metadata.otherPartyKeys(), metadata.cryptoSignatures(), accessor));
+                }
+            } else {
+                accessor.setSigMeta(noneAvailable());
+            }
+            // Prevent the mono-service worfklow from rationalizing sigs
+            accessor.setExpandedSigStatus(preHandleStatus);
+            accessor.setLinkedRefs(new LinkedRefs());
+            return accessor;
+        } catch (InvalidProtocolBufferException e) {
+            throw new AssertionError("Not implemented");
```

Correct error message?

---

More than that, it might be better to throw a matching exception from PBJ rather than fail the JVM with an Error.